docs/bookstore/page-0.5/pg0.5-env-setup.md

```diff
@@ -0,0 +1,229 @@
+- You have your front end application deployed as kubernetes deployment with port-forwarding to localhost:3000
+- You have your Node.js application deployed as kubernetes deployment with port-forwarding to localhost:8080
+
+We will be fulfilling each requirement with the order above.
+
+## **File Structure**
+
+![Image](images/image21.png)
+
+The code for the sample app is in `docs/code-samples/eventing/bookstore-sample-app`
+
+Under bookstore-sample-app folder, there are 2 folders:
+
+**/solution**: this folder contains all the yaml files, and the code you needed. Check it when you get stuck.
+
+**/start**: this folder contains the necessary files for you to get started. Write all the configuration files yourself following the tutorial!
+
+![Image](images/image16.png)
+
+!!! tip
+
+    Kuack suggests you to start from **/start**, write all the configuration files as you go over the tutorial, and check solutions when you get stuck.
+
+
+## **Shortcut**
+
+![Image](images/image10.png)
+
+Running `docs/code-samples/eventing/bookstore-sample-app/start/setup.sh` will automatically complete all tasks in this section.
+
+!!! warning
+
+    However, if you are not familiar with the process, we recommend reviewing the steps below.
+
+
+
+## **Instructions**
+
+### **Task 1: Set Up a Running Kubernetes Cluster with Knative Installed**
+
+![Image](images/image13.png)
+
+Please follow the instructions here [https://knative.dev/docs/install/](https://knative.dev/docs/install/) to spin up your cluster!
+
+
+
+???+ success "Verify"
+
+    You should see the pods in knative-eventing and knative-serving are running before proceeding forward.
+
+    ![Image](images/image2.png)
+
+#### **Extra Step for Minikube Users:**
+
+![Image](images/image3.png)
+
+Attention! People always forget this! If you're not using the Knative Quick Start, manually set up the tunnel to connect to LoadBalancer services.
+
+Run the following command and keep the terminal open:
+
+```shell
+minikube tunnel
+```
+
+???+ success "Verify"
+    If there aren't any error messages, it means you have set up the tunnel successfully.
+
+### **Task 2: Running the Bookstore Web App**
+
+![Image](images/image12.png)
+
+The Next.js frontend app is located in the `docs/code-samples/eventing/bookstore-sample-app/start/frontend` folder.
+
+Ensure that port 3000 on your local machine is not being used by another application.
+
+#### **Deploy the Frontend App**
+
+You can either [build the image locally](https://docs.docker.com/get-started/02_our_app/) or use our pre-built image. If you are using the pre-built image, you can proceed to the next step.
+
+When ready, run the following command to deploy the frontend app:
+
+```shell
+kubectl apply -f config/100-front-end-deployment.yaml
+```
+
+![Image](images/image11.png)
+
+This command will pull the image and deploy it to your cluster as a Deployment. It will also expose it as a LoadBalancer to receive external traffic.
```

Comment:
```suggestion
This will create the Deployment and expose it with a Service of type LoadBalancer to receive external traffic:

![Image](images/image11.png)
```